### ISSUE TITLE
Extracting SCI_COR_Subarray from tar-files causes IndexError exception

### ISSUE THREAD

**Issue by mathiasbeck** [pmaxted/pycheops#111]
Dear Pierre,

I started to use pycheops and ran into my first bug. I think it is easy to fix as I patched my local version of pycheops to better understand ...

When I download a GTO data set from DACE via the Dataset(file_key) and subsequently run a get_subarray() I get an exception about an invalid index.

The problem is linked to the fact that the tar-file contains a SCI_COR_SubArray which is actually being used for the extraction. For the SCI_COR_SubArray, the the meta-data is in HDU-2 and not as for the SCI_RAW_SubArray in HDU-9. By the way, you correctly handle this if the FITS-file is already extracted ...

So you need to catch the IndexError exception in the same way as a few lines above ...

```
            with tar.extractfile(datafile[0]) as fd:
                hdul = fits.open(fd)
                cube = hdul[1].data
                hdr = hdul[1].header
                # Meta can be in extention 9 or 2, so...
                try:
                    meta = Table.read(hdul[9])
                except IndexError:
                    meta = Table.read(hdul[2])
                hdul.writeto(subPath)

```

Thanks a lot for the opportunity to look at some source code after so many years ;-)

Kind regards, Mathias

**Comment by ThomasGWilson** [comment]
Hi Mathias,

I've caught this bug a few hours ago too and have already created a patch for it. I'm sure it'll be included in the next release.

Thanks,
Tom

**Comment by mathiasbeck** [comment]
Dear Tom,

That was really fast. Thanks a lot !

I think that it would be even more robust to replace the numerical indexes to retrieve the HDUs by their names like 'SCI_COR_ImageMetaData'. But that requires a few more code changes ...

Thanks again, Mathiias

**Comment by pmaxted** [comment]
Fixed but left open because the suggestion to find the index automatically from the extension name should be implemented.

**Comment by ThomasGWilson** [comment]
Getting the HDUs by extension names should be included in the next release.

**Comment by pmaxted** [comment]
Fixed in version 0.8.0
